### PR TITLE
Archive updates

### PR DESCRIPTION
**Pull request by johnpryan**
Update 6/6: This adds a new package, `dart_sdk_archive` that can be used to make updates to the `download_archive.dart` script's functionality.

fixes dart-lang/site-shared#96
fixes dart-lang/site-shared#97
fixes dart-lang/site-www#75 

This doesn't affect #89 - I noticed there was an `install.js` script that runs some google analytics but I haven't looked into whether or not it's working anymore

### CODE REVIEW

File: src/tools/sdk/archive/web/download_archive.dart [hunk 1]
```diff
@@ -145,22 +146,19 @@ const Map<String, String> suffixMap = const {
...
-      const PlatformVariant('64-bit', const ['Dart SDK', 'Dartium']),
-    ],
-  'Linux':
-    const [
-      const PlatformVariant('ARMv7', const ['Dart SDK']),
-      const PlatformVariant('ARMv8 (ARM64)', const ['Dart SDK']),
-      const PlatformVariant('32-bit', const ['Dart SDK', 'Dartium']),
-      const PlatformVariant('64-bit', const ['Dart SDK', 'Dartium']),
+  'Mac': const [
+    const PlatformVariant('32-bit', const ['Dart SDK', 'Dartium']),
```
Comment (kwalrath) [inline src/tools/sdk/archive/web/download_archive.dart]: We could get rid of these consts, no?

File: src/tools/sdk/archive/web/download_archive.dart [hunk 2]
```diff
@@ -223,12 +221,12 @@ void addVersion(String channel, Map<String, String> version) {
...
         ..classes.add('muted'));
 
-      row.addCell()..text = name;
+      row.addCell()..text = archiveMap[name];
       row.addCell()
         ..classes.add('nowrap')
         ..text = platformVariant.architecture;
-      List<String> possibleArchives = ['Dart SDK', 'Dartium'];
-      TableCellElement c = row.addCell()..classes.add('archives');
+      var possibleArchives = <String>['Dart SDK', 'Dartium'];
```
Comment (kwalrath) [inline src/tools/sdk/archive/web/download_archive.dart]: Shouldn't the `<String>` be inferred?

Comment (johnpryan) [reply src/tools/sdk/archive/web/download_archive.dart]: good catch, fixed

File: src/tools/sdk/archive/web/download_archive.dart (hunk 1 above)
Comment (johnpryan) [reply src/tools/sdk/archive/web/download_archive.dart]: fixed, thanks

**Comment by johnpryan** [comment]
@domesticmouse could you take a look at this if you have time?

I also need to call out that this uses a git dependency on `package:sdk_builds` until https://github.com/kevmoo/sdk_builds.dart/pull/1 is merged

**Comment by kwalrath** [comment]
I've staged this:
https://dartlang-org-staging-0.firebaseapp.com/tools/sdk/archive
(compare to https://dart.dev/tools/sdk/archive)

File: src/tools/sdk/dart_sdk_archive/analysis_options.yaml [hunk 3]
```diff
@@ -0,0 +1 @@
+include: package:pedantic/analysis_options.yaml
```
Comment (domesticmouse) [inline src/tools/sdk/dart_sdk_archive/analysis_options.yaml]: include: package:pedantic/analysis_options.1.7.0.yaml to not get travis build failures when a new version of pedantic is released

File: src/tools/sdk/dart_sdk_archive/web/styles.css [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+}
+
+#output {
+  padding: 20px;
+  text-align: center;
+}
+
+.hidden {
+  display: none;
+}
```
Comment (domesticmouse) [inline src/tools/sdk/dart_sdk_archive/web/styles.css]: nit: add newline

**Comment by johnpryan** [comment]
@domesticmouse the sdk_builds.dart package isn't going to be maintained so I added it to this project instead

fyi @kevmoo

**Comment by johnpryan** [comment]
@domesticmouse @kwalrath PTAL

**Comment by kwalrath** [comment]
@jamesderlin do you want to take a look at the staged site? It’s at https://kw-www-dartlang-1.firebaseapp.com/tools/sdk/archive.

**Comment by jamesderlin** [comment]
Much better, thanks!  I would order x64 before ia32 and ARMv8 before ARMv7. (Do we have download statistics that we can check to verify my presumption that 64-bit packages are more frequently downloaded than 32-bit ones?)

**Comment by kwalrath** [comment]
@johnpryan we talked about analytics for downloads… Were you able to get anywhere with those? If nothing else, I can check with the people who run the download server.

**Comment by kwalrath** [comment]
I’m going to commit this change; we can tweak the order later.

**Comment by johnpryan** [comment]
@kwalrath no, this doesn't add analytics to this page, unfortunately.

**Pull request merged**